datalad/support/gitrepo.py

```diff
@@ -2899,6 +2898,25 @@ def get_content_info(self, paths=None, ref=None, untracked='all'):
             if "fatal: Not a valid object name" in str(exc):
                 raise ValueError("Git reference '{}' invalid".format(ref))
             raise
+        lgr.debug('Done query repo: %s', cmd)
+
+        if ref:
+            def _read_symlink_target_from_catfile(lines):
+                # it is always the second line, all checks done upfront
+                header = lines.readline()
+                if header.rstrip().endswith('missing'):
+                    # something we do not know about, should not happen
+                    # in real use, but guard against to avoid stalling
+                    return ''
+                return lines.readline().rstrip()
```

Comment:
This function is being called with only with targets where `inf['type'] == 'symlink'`, so we know that we can expect two lines (unlike the general cat-file --batch output).  OK.

Reply:
Yes, correct.